static/panes/tool-input-view.js

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) 2021, Tom Ritter
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+'use strict';
+
+var FontScale = require('../fontscale');
+var monaco = require('monaco-editor');
+var _ = require('underscore');
+var $ = require('jquery');
+var ga = require('../analytics');
+var monacoConfig = require('../monaco-config');
+var local = require('../local');
+
+require('../modes/asm-mode');
+
+function ToolInputView(hub, container, state) {
+    state = state || {};
+    this.container = container;
+    this.eventHub = hub.createEventHub();
+    this.domRoot = container.getElement();
+    this.domRoot.html($('#tool-input').html());
+    this.source = state.source || '';
+    this._currentDecorations = [];
+    var root = this.domRoot.find('.monaco-placeholder');
+
+    this.settings = JSON.parse(local.get('settings', '{}'));
+
+    this.editor = monaco.editor.create(root[0], monacoConfig.extendConfig({
+        value: '',
+        language: 'plaintext',
+        readOnly: false,
+        glyphMargin: true,
+    }));
+
+    this._toolId = state.toolId;
+    this._toolName = state.toolName;
+    this._compilerId = state.compilerId;
+    this._editorId = state.editorId;
+
+    this.awaitingInitialResults = false;
+
+    this.initButtons(state);
+    this.initCallbacks();
+
+    this.setTitle();
+    this.onSettingsChange(this.settings);
+    this.eventHub.emit('toolInputViewOpened', this._toolid);
+    ga.proxy('send', {
```

Comment:
Thanks: we don't get many users choosing to enable google analytics but this will let us guess how many folks use this new feature! Just ping me on Discord and I can share the analytics with you.